AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Subscribe/Support/Model+Sort.swift

```diff
@@ -0,0 +1,156 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import CoreGraphics
+
+extension Array where Element: Model {
+    mutating func sortModels(by sortBy: QuerySortBy, modelSchema: ModelSchema) {
+        sort { modelSchema.comparator(model1: $0, model2: $1, sortBy: sortBy) }
+    }
+}
+
+enum ModelValueCompare<T: Comparable> {
+    case bothNil
+    case leftNil(value2: T)
+    case rightNil(value1: T)
+    case values(value1: T, value2: T)
+    case unknown
+
+    init(value1Optional: T?, value2Optional: T?) {
+        if value1Optional == nil && value2Optional == nil {
+            self = .bothNil
+        } else if value1Optional == nil, let value2 = value2Optional {
+            self = .leftNil(value2: value2)
+        } else if value2Optional == nil, let value1 = value1Optional {
+            self = .rightNil(value1: value1)
+        } else if let value1 = value1Optional, let value2 = value2Optional {
+            self = .values(value1: value1, value2: value2)
+        } else {
+            self = .unknown
+        }
+    }
+
+    /// Treat `nil` as less than non `nil`values, so return asecnding when left side is `nil` and descending when right
+    /// is nil.
+    func sortComparator(sortOrder: QuerySortOrder) -> Bool {
+        switch self {
+        case .bothNil, .leftNil:
+            return sortOrder == .ascending
+        case .rightNil:
+            return sortOrder == .descending
+        case .values(let value1, let value2):
+            return sortOrder == .ascending ? value1 < value2 : value1 > value2
+        case .unknown:
+            return false
+        }
+    }
+}
+
+extension ModelSchema {
+    // swiftlint:disable:next cyclomatic_complexity
```

Comment:
nit: would breaking down the below function in smaller functions help to reduce the cyclomatic complexity?